src/CLI/OutputGenerator.cs

```diff
@@ -0,0 +1,152 @@
+        {
+            bool failedToComplete = errorCollector.ParameterErrors.Any() || errorCollector.Exceptions.Any() || (scanResults == null);
+
+            ShowBanner(options, failedToComplete ? VerbosityLevel.Quiet : VerbosityLevel.Default);
+            if (failedToComplete)
+            {
+                ShowExecutionErrors(errorCollector);
+            }
+            else
+            {
+                ShowScanResults(options, scanResults);
+            }
+        }
+
+        public void ShowBanner(IOptions options)
+        {
+            ShowBanner(options, VerbosityLevel.Default);
+        }
+
+        private void ShowBanner(IOptions options, VerbosityLevel miniumVerbosity)
+        {
+            if (options == null) throw new ArgumentNullException(nameof(options));
+
+            if(!_bannerHasBeenShown && options.VerbosityLevel >= miniumVerbosity)
+            {
+                string version = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location).ProductVersion;
+                _writer.WriteLine("Axe.Windows Accessibility Scanner (version {0})", version);
+
+                bool haveProcessName = options.ProcessName != IProcessHelper.InvalidProcessName;
+                bool haveProcessId = options.ProcessId != IProcessHelper.InvalidProcessId; 
+
+                if (haveProcessName || haveProcessId)
+                {
+                    _writer.Write("Scan Target:");
+
+                    if (haveProcessName)
+                    {
+                        _writer.Write(" Process Name = {0}", options.ProcessName);
+                    }
+                    if (haveProcessName && haveProcessId)
+                    {
+                        _writer.Write(",");
+                    }
+                    if (haveProcessId)
+                    {
+                        _writer.Write(" Process ID = {0}", options.ProcessId);
+                    }
+                    _writer.WriteLine();
+                }
+                if (!string.IsNullOrEmpty(options.ScanId))
+                {
+                    _writer.WriteLine("Scan Id = {0}", options.ScanId);
+                }
+
+                _bannerHasBeenShown = true;
+            }
+        }
+
+        private void ShowExecutionErrors(IErrorCollector errorCollector)
+        {
+            foreach (string parameterError in errorCollector.ParameterErrors)
+            {
+                _writer.WriteLine("Parameter Error: {0}", parameterError);
+            }
+
+            foreach (Exception exception in errorCollector.Exceptions)
+            {
+                _writer.WriteLine("The following exception was caught: {0}", exception);
+            }
+        }
+
+        private void ShowScanResults(IOptions options, ScanResults scanResults)
+        {
+            if (options.VerbosityLevel == VerbosityLevel.Quiet)
+            {
+                return;
+            }
+
+            if (scanResults.ErrorCount == 1)
```

Comment:
Perhaps extract to a function called WriteErrorCount?

Reply:
Done